src/CRISP.BackendChallenge/Controllers/EmployeeController.cs

```diff
@@ -7,33 +7,266 @@
+            var employees = _employeeRepository.GetAll();
+
+            var result = employees.Select(ToEmployeeResponse);
+
+            return Ok(result);
+        }
+        catch (ArgumentNullException e)
+        {
+            _logger.LogDebug(":: There are no Employees in the Database: {EMessage}", e.Message);
+            return NotFound(new {error = "No Employees Found in Database"});
+        }
+        catch (Exception e)
+        {
+            _logger.LogError(":: Error occurred while getting all employees: {EMessage}", e.Message);
+            return CreateInternalServerError();
+        }
+    }
+
+    [HttpGet]
+    [Route("{id:int}")]
+    public IActionResult GetById(int id)
+    {
+        _logger.LogDebug(":: Performing {MethodName} with ID {Id}", nameof(GetById), id);
+        try
+        {
+            var employee = _employeeRepository.GetById(id);
+            return Ok(ToEmployeeResponse(employee));
+        }
+        catch (ArgumentNullException e)
+        {
+            _logger.LogDebug(":: Id Not Found: {Id}, {EMessage}", id, e.Message);
+            return CreateNotFound(id);
+        }
+        catch (Exception e)
+        {
+            _logger.LogError(":: Error occurred while getting employee ID: {Id} {EMessage}", id, e.Message);
+            return CreateInternalServerError();
+        }
+    }
+    
+    
+    [HttpPost]
+    [Route("createEmployee")]
+    public IActionResult AddEmployee([FromBody] EmployeeRequest employeeRequest)
+    {
+        _logger.LogDebug(":: Performing {MethodName}", nameof(AddEmployee));
+        
+        if (string.IsNullOrWhiteSpace(employeeRequest.Name))
+        {
+            _logger.LogError(":: Invalid EmployeeRequest provided");
+            return CreateBadRequest("Creating an employee", "Name field required");
+        }
+        if (employeeRequest.Department == null)
+        {
+            _logger.LogError(":: Invalid EmployeeRequest provided");
+            return CreateBadRequest("Creating an employee", "Department field required");
+        }
+        
+        try
+        {
+            
+            var employee = new Employee
+            {
+                Name = employeeRequest.Name,
+                Department = (Department)employeeRequest.Department,
+                StartDate = employeeRequest.StartDate,
+                EndDate = employeeRequest.EndDate
+            };
+
+            _employeeRepository.Add(employee);
+            _employeeRepository.Save();
+
+            if (employeeRequest.LoginDates != null || employeeRequest.LoginDates!.Any())
+            {
+                foreach (var login in employeeRequest.LoginDates!.Select(l => new Login()
+                         {
+                             EmployeeId = employee.Id,
+                             LoginDate = l
+                         }))
+                {
+                    _loginRepository.Add(login);
+                }
+                _loginRepository.Save();
+            }
```

Comment:
This code is unnecessary - the `Employee` entity model already has a navigation property linking it to the `Logins` table. If you attach them to the entity upon creation, the ORM will automatically add them to the necessary table:

```cs
var employee = new Employee
{
    Name = employeeRequest.Name,
    Department = (Department)employeeRequest.Department,
    StartDate = employeeRequest.StartDate,
    EndDate = employeeRequest.EndDate,
    Logins = employeeRequest.LoginDates?.Select(loginDate => new Login {
      EmployeeId = employee.Id,
      LoginDate = loginDate
    });
};

_employeeRepository.Add(employee);
_employeeRepository.Save(); // Will add the Employee, as well as every Login
```

Reply:
Makes sense, updated to reflect as such

Reply:
This comment still applies in a few spots.. I still see a lot of logic in the controller. The controller should not know about internal application logic